internal/generateOmgIdl.ts

```diff
@@ -0,0 +1,111 @@
+import { FoxglovePrimitive, FoxgloveSchema } from "./types";
+
+function primitiveToIdl(type: Exclude<FoxglovePrimitive, "time" | "duration">) {
+  switch (type) {
+    case "bytes":
+      return "sequence<uint8>";
+    case "string":
+      return "string";
+    case "boolean":
+      return "boolean";
+    case "float64":
+      return "double";
+    case "uint32":
+      return "uint32";
+  }
+}
+
+export const TIME_IDL = `struct Time {
```

Comment:
should we namespace Time and Duration to `foxglove::`? It looks like we do that for flatbuffers